pallets/vcu/src/lib.rs

```diff
@@ -390,18 +392,15 @@ decl_module! {
 			// check whether asset generated VCU exists or not
 			ensure!(AssetsGeneratingVCU::<T>::contains_key(&account_id, &signer), Error::<T>::AssetGeneratedVCUNotFound);
 
-			let content: Vec<u8> = AssetsGeneratingVCU::<T>::get(&account_id, &signer);
-			let total_shares = Self::json_get_value(content.clone(),"numberOfShares".as_bytes().to_vec());
-
-			ensure!((str::parse::<u32>(sp_std::str::from_utf8(&total_shares).unwrap()).unwrap() + number_of_shares) <= 10000 , Error::<T>::TooManyNumberofShares);
-
 			AssetsGeneratingVCUShares::<T>::try_mutate(&recipient, &agv_id, |share| -> DispatchResult {
 				*share = number_of_shares;
 				Ok(())
 			})?;
 
 			AssetsGeneratingVCUSharesMinted::<T>::try_mutate(&account_id, &signer, |share| -> DispatchResult {
-				*share = number_of_shares;
+				let total_sh = share.checked_add(number_of_shares).ok_or(Error::<T>::Overflow)?;
```

Comment:
You should not check a static 10000 but the maximum number of shares stored in the AVG